components/tools/OmeroWeb/omeroweb/webgateway/views.py

```diff
@@ -1463,6 +1474,60 @@ def get_thumb_url(iid):
 
 @login_required()
 @jsonp
+def get_thumbnails_json(request, w=None, conn=None, **kwargs):
+    """
+    Returns base64 encoded jpeg with the rendered thumbnail for images
+    'id'
+
+    @param request:     http request
+    @param w:           Thumbnail max width. 96 by default
+    @return:            http response containing base64 encoded thumbnails
+    """
+    if w is None:
+        w = 96
+    image_ids = get_longs(request, 'id')
+    logger.debug("Image ids: %r" % image_ids)
+    if len(image_ids) > settings.THUMBNAILS_BATCH:
+        return HttpJavascriptResponseServerError(
+            'Max %s thumbnails at a time.' % settings.THUMBNAILS_BATCH)
+    thumbnails = conn.getThumbnailSet([rlong(i) for i in image_ids], w)
+    rv = dict()
+    for i in image_ids:
+        try:
+            t = thumbnails[i]
+            if len(t) > 0:
+                # replace thumbnail urls by base64 encoded image
+                rv[i] = ("data:image/jpeg;base64,%s" % base64.b64encode(t))
+            else:
+                rv[i] = None
+        except Exception:  # TypeError, KeyError
+            logger.error(traceback.format_exc())
+    return rv
+
+
+@login_required()
+@jsonp
+def get_thumbnail_json(request, iid, w=None, h=None, conn=None, _defcb=None,
+                       **kwargs):
+    """
+    Returns an HttpResponse wrapped jpeg with the rendered thumbnail for image
+    'iid'
+
+    @param request:     http request
+    @param iid:         Image ID
+    @param w:           Thumbnail max width. 96 by default
+    @param h:           Thumbnail max height
+    @return:            http response containing base64 encoded thumbnail
+    """
+    jpeg_data = _render_thumbnail(
+        request=request, iid=iid, w=w, h=h,
+        conn=conn, _defcb=_defcb, **kwargs)
+    rv = "data:image/jpeg;base64,%s" % base64.b64encode(jpeg_data)
+    return rv
```

Comment:
It meant to be inline with render_thumbnail to let people easily switch without much of the code changes. This will have an impact on `refreshThumbnails`, my idea was to reduce number of changes.